app/services/SubmissionService.scala

```diff
@@ -94,21 +93,30 @@ class SubmissionService @Inject()(
 
   def submit(providerId: String, answers: MovementAnswers)(implicit hc: HeaderCarrier): Future[ConsignmentReferences] = {
     val cache = Cache(providerId, answers)
```

Comment:
There's no need to create a `Cache` here, all you use it for is `cache.answers.type`, you might as well just store the `answers.type` in a variable.

That sorts the need to rename `cache` to `cacheRepository`

Reply:
What I'm saying here is `submit` & `movementAuditType` can become

```
def submit(providerId: String, answers: MovementAnswers)(implicit hc: HeaderCarrier): Future[ConsignmentReferences] = {
    val data = movementBuilder.createMovementExchange(providerId, answers)
    val timer = metrics.startTimer(answers.`type`)

    def performSideEffects(result: String)(implicit hc: HeaderCarrier): Unit = {
      metrics.incrementCounter(answers.`type`)
      auditService.auditMovements(data, result, movementAuditType(answers.`type`))
      timer.stop()
    }

    auditService.auditAllPagesUserInput(answers)
    connector
      .submit(data)
      .map(_ => cacheRepository.removeByProviderId(providerId))
      .map(_ => data.consignmentReference)
      .andThen {
        case Success(_) => performSideEffects(success)
        case Failure(_) => performSideEffects(failed)
      }

  }

  private def movementAuditType(`type`: JourneyType): AuditType.Value = `type` match {
    case JourneyType.ARRIVE               => AuditType.AuditArrival
    case JourneyType.RETROSPECTIVE_ARRIVE => AuditType.AuditRetrospectiveArrival
    case JourneyType.DEPART               => AuditType.AuditDeparture
  }
```

Then you don't need to create a `Cache`